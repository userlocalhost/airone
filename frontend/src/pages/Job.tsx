import { Box, Button, Theme, Typography } from "@mui/material";
import { makeStyles } from "@mui/styles";
import React, { FC } from "react";
import { Link } from "react-router-dom";
import { useAsync } from "react-use";

import { topPath } from "../Routes";
import { AironeBreadcrumbs } from "../components/common/AironeBreadcrumbs";
import { Loading } from "../components/common/Loading";
import { JobList } from "../components/job/JobList";
import { getJobs } from "../utils/AironeAPIClient";

const useStyles = makeStyles<Theme>((theme) => ({
  button: {
    margin: theme.spacing(1),
  },
  entityName: {
    margin: theme.spacing(1),
  },
}));

export const Job: FC = () => {
  const classes = useStyles();

  const jobs = useAsync(async () => {
    const resp = await getJobs();
    return await resp.json();
  }, []);

  return (
    <Box className="container-fluid">
      <AironeBreadcrumbs>
        <Typography component={Link} to={topPath()}>
          Top
        </Typography>
        <Typography color="textPrimary">ジョブ一覧</Typography>
      </AironeBreadcrumbs>

      <Box className="row">
        <Box className="col">
          <Box className="float-left">
            <Button
              className={classes.button}
              variant="outlined"
              color="primary"
            >
              全件表示
            </Button>
          </Box>
          <Box className="float-right"></Box>
        </Box>
      </Box>

      {jobs.loading ? <Loading /> : <JobList jobs={jobs.value} />}
    </Box>
  );
};
