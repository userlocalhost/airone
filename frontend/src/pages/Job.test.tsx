/**
 * @jest-environment jsdom
 */

import {
  render,
  waitForElementToBeRemoved,
  screen,
} from "@testing-library/react";
import React from "react";

import { TestWrapper } from "../utils/TestWrapper";

import { Job } from "./Job";

afterEach(() => {
  jest.clearAllMocks();
});

test("should match snapshot", async () => {
  const groups = [
    {
      id: 1,
      operation: 1,
      status: 1,
      passed_time: "2022-01-01 00:00:00",
      created_at: "2022-01-01 00:00:00",
      note: "note",
      target: {
        name: "target1",
      },
    },
    {
      id: 2,
      operation: 2,
      status: 2,
      passed_time: "2022-01-01 00:00:00",
      created_at: "2022-01-01 00:00:00",
      note: "note",
      target: {
        name: "target2",
      },
    },
  ];

  /* eslint-disable */
  jest.spyOn(require("../utils/AironeAPIClient"), "getJobs").mockResolvedValue({
    json() {
      return Promise.resolve(groups);
    },
  });
  /* eslint-enable */

  // wait async calls and get rendered fragment
  const result = render(<Job />, {
    wrapper: TestWrapper,
  });
  await waitForElementToBeRemoved(screen.getByTestId("loading"));

  expect(result).toMatchSnapshot();
});
