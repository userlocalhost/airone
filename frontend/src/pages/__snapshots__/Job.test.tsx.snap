// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid MuiBox-root css-0"
      >
        <nav
          aria-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root makeStyles-breadcrumbs-3 css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                href="/new-ui/"
              >
                Top
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
              >
                ジョブ一覧
              </p>
            </li>
          </ol>
        </nav>
        <div
          class="row MuiBox-root css-0"
        >
          <div
            class="col MuiBox-root css-0"
          >
            <div
              class="float-left MuiBox-root css-0"
            >
              <button
                class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root makeStyles-button-1 css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                全件表示
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="float-right MuiBox-root css-0"
            />
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
        >
          <table
            class="MuiTable-root css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    対象エントリ
                  </p>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    操作
                  </p>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    状況
                  </p>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    実行時間
                  </p>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    実行日時
                  </p>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                  scope="col"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    備考
                  </p>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                  scope="col"
                />
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    target1
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    1
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    1
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    2022-01-01 00:00:00
                     s
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    2022-01-01 00:00:00
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    note
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                >
                  <ul
                    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                  >
                    <li
                      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                    >
                      <a
                        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-6 css-sghohy-MuiButtonBase-root-MuiButton-root"
                        href="/jobs/1/rerun"
                        tabindex="0"
                      >
                        Re-run
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                    >
                      <a
                        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-6 css-zcbmsk-MuiButtonBase-root-MuiButton-root"
                        href="/jobs/1/cancel"
                        tabindex="0"
                      >
                        Cancel
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                    </li>
                  </ul>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    target2
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    2
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    2
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    2022-01-01 00:00:00
                     s
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    2022-01-01 00:00:00
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    note
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                >
                  <ul
                    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                  >
                    <li
                      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                    >
                      <a
                        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-6 css-sghohy-MuiButtonBase-root-MuiButton-root"
                        href="/jobs/2/rerun"
                        tabindex="0"
                      >
                        Re-run
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                    >
                      <a
                        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-6 css-zcbmsk-MuiButtonBase-root-MuiButton-root"
                        href="/jobs/2/cancel"
                        tabindex="0"
                      >
                        Cancel
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                    </li>
                  </ul>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid MuiBox-root css-0"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root makeStyles-breadcrumbs-3 css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              href="/new-ui/"
            >
              Top
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
            >
              ジョブ一覧
            </p>
          </li>
        </ol>
      </nav>
      <div
        class="row MuiBox-root css-0"
      >
        <div
          class="col MuiBox-root css-0"
        >
          <div
            class="float-left MuiBox-root css-0"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root makeStyles-button-1 css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              全件表示
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="float-right MuiBox-root css-0"
          />
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  対象エントリ
                </p>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  操作
                </p>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  状況
                </p>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  実行時間
                </p>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  実行日時
                </p>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  備考
                </p>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              />
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  target1
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  1
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  1
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  2022-01-01 00:00:00
                   s
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  2022-01-01 00:00:00
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  note
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <ul
                  class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                >
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <a
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-6 css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="/jobs/1/rerun"
                      tabindex="0"
                    >
                      Re-run
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <a
                      class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-6 css-zcbmsk-MuiButtonBase-root-MuiButton-root"
                      href="/jobs/1/cancel"
                      tabindex="0"
                    >
                      Cancel
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </li>
                </ul>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  target2
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  2
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  2
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  2022-01-01 00:00:00
                   s
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  2022-01-01 00:00:00
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  note
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <ul
                  class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                >
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <a
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-6 css-sghohy-MuiButtonBase-root-MuiButton-root"
                      href="/jobs/2/rerun"
                      tabindex="0"
                    >
                      Re-run
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <a
                      class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-6 css-zcbmsk-MuiButtonBase-root-MuiButton-root"
                      href="/jobs/2/cancel"
                      tabindex="0"
                    >
                      Cancel
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </li>
                </ul>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
